pkg/localstore/migration.go

```diff
@@ -101,3 +105,265 @@ func getMigrations(currentSchema, targetSchema string, allSchemeMigrations []mig
 	}
 	return migrations, nil
 }
+
+// migrateYuj removes all existing database content, unless
+// pinned content is detected, in which case it aborts the
+// operation for the user to resolve.
+func migrateYuj(db *DB) error {
+	pinIndex, err := db.shed.NewIndex("Hash->PinCounter", shed.IndexFuncs{
+		EncodeKey: func(fields shed.Item) (key []byte, err error) {
+			return fields.Address, nil
+		},
+		DecodeKey: func(key []byte) (e shed.Item, err error) {
+			e.Address = key
+			return e, nil
+		},
+		EncodeValue: func(fields shed.Item) (value []byte, err error) {
+			b := make([]byte, 8)
+			binary.BigEndian.PutUint64(b[:8], fields.PinCounter)
+			return b, nil
+		},
+		DecodeValue: func(keyItem shed.Item, value []byte) (e shed.Item, err error) {
+			e.PinCounter = binary.BigEndian.Uint64(value[:8])
+			return e, nil
+		},
+	})
+	if err != nil {
+		return err
+	}
+	c, err := pinIndex.Count()
+	if err != nil {
+		return fmt.Errorf("get pin index count: %w", err)
+	}
+	if c > 0 {
+		return errors.New("pinned content detected. bee v0.6.0 cannot migrate a store with pinned content. please refer to the release notes for more information.")
```

Comment:
This is very important information to the user, but contains a rather obscure and frightening message. I am making a suggestion below, mostly to give the user better instructions.

I would suggest to:
1. log an error message, saying: `failed to update your node due to the existence of pinned content. Please refer to the release notes on how to safely migrate your pinned content. Your localstore location is <path_to_localstore>`
2. return an error: `node cannot boot up with an outdated database scheme`